File: Demo/app/src/main/swift/UnitEmitter.swift
//
//  UnitEmitter.swift
//  SwiftAndroidApp
//
//  Created by Alsey Coleman Miller on 7/13/25.
//

import JavaKit

/// Bridge from Swift to Kotlin Coroutines
@JavaClass("com.pureswift.swiftandroid.UnitEmitter")
open class UnitEmitter: JavaObject {
    
    @JavaMethod
    @_nonoverride public convenience init(environment: JNIEnvironment? = nil)
    
    @JavaMethod
    func emit()
}

File: Demo/app/src/main/java/com/pureswift/swiftandroid/UnitEmitter.kt
package com.pureswift.swiftandroid

import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.SharedFlow

class UnitEmitter() {

    private val _flow = MutableSharedFlow<Unit>(extraBufferCapacity = 64)
    val flow: SharedFlow<Unit> get() = _flow

    fun emit() {
        //println("Emit")
        _flow.tryEmit(Unit)
    }
}